streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -192,7 +204,8 @@ public void initStoresFromCheckpointedOffsets() {
             }
 
             checkpointFile.delete();
-        } catch (final IOException e) {
+        } catch (final IOException | RuntimeException e) {
```

Comment:
I realized that some parsing exceptions are just runtime exceptions so catching both here.